lib/config/fileconf.go

```diff
@@ -1523,6 +1523,8 @@ type GCPMatcher struct {
 type AccessGraphSync struct {
 	// AWS is the AWS configuration for the AccessGraph Sync service.
 	AWS []AccessGraphAWSSync `yaml:"aws,omitempty"`
+	// Azure is the Azure configuration for the AccessGraph Sync service.
+	Azure []AccessGraphAzureSync `yaml:"azure,omitempty"`
```

Comment:
Do you mean using the [discoveryconfig_service.proto](https://github.com/gravitational/teleport/blob/master/api/proto/teleport/discoveryconfig/v1/discoveryconfig_service.proto) to apply configuration to pull resources for the access graph? That sounds great to me--i.e. having a unified interface for discovering resources with some additional config about where those resources go, rather than duplicating config and reporting. @tigrato thoughts? We can make a separate issue for this and decouple it from this Azure integration feature.